src/micro_http/src/server.rs

```diff
@@ -720,6 +757,38 @@ mod tests {
         let mut buf: [u8; 120] = [0; 120];
         sockets[MAX_CONNECTIONS].read_exact(&mut buf).unwrap();
         assert_eq!(&buf[..], SERVER_FULL_ERROR_MESSAGE);
+        assert_eq!(server.connections.len(), 10);
+        {
+            // Drop this stream.
+            let _refused_stream = sockets.pop().unwrap();
+        }
+        assert_eq!(server.connections.len(), 10);
```

Comment:
Isn't this assert redundant since dropping a refused stream would not change the actual list of connections checked at line 760?

Reply:
The assertion at line 760 is to check that the refused stream wasn't added to the active connection map in the server.

The check at line 765 is to check that the refused stream didn't somehow replace another previously accepted active connection. If this was the case, then closing the socket would make the number of connections drop to 9, but the assertion verifies it to not change.

Reply:
Understood, thanks!